src/action.js

```diff
@@ -0,0 +1,182 @@
+    const head = `patcher-updates-${dependency}`
+    const title = `[Patcher] Update ${dependency}`
+    const commitMessage = "Update dependencies using Patcher"
+    const commitAuthor = "Grunty"
+    const commitEmail = "grunty@gruntwork.io"
+
+    const body = `
+Updated the \`${dependency}\` dependency using Patcher.
+
+### Update summary
+\`\`\`yaml
+${output}
+\`\`\`
+`
+
+    await exec.exec("git", ["config", "user.name", commitAuthor])
+    await exec.exec("git", ["config", "user.email", commitEmail])
+    await exec.exec("git", ["add", "."])
+    await exec.exec("git", ["commit", "-m", commitMessage])
+    await exec.exec("git", ["checkout", "-b", head])
+
+    const context = github.context;
+
+    await exec.exec("git", ["push", "-f", `https://${ghToken}@github.com/${context.repo.owner}/${context.repo.repo}.git`])
+
+    const octokit = new github.getOctokit(ghToken);
+
+    const repoDetails = await octokit.rest.repos.get({ ...context.repo });
+    const base = repoDetails.data.default_branch;
+    core.debug(`Base branch is ${base}. Opening the PR against it.`)
+
+    try {
+        await octokit.rest.pulls.create({ ...context.repo, title, head, base, body, });
+    } catch (error) {
+        if (error.message?.includes(`A pull request already exists for`)) {
+            core.error(`A pull request for ${head} already exists. The branch was updated.`)
+        } else {
+            throw error
+        }
+    }
+}
+
+export async function downloadRelease(owner, repo, tag, ghToken) {
+    core.info(`Downloading Patcher version ${patcherVersion}`);
+
+    const octokit = new github.getOctokit(ghToken);
+
+    const getReleaseUrl = await octokit.rest.repos.getReleaseByTag({ owner, repo, tag })
+
+    const re = new RegExp(`${osPlatform()}.*amd64`)
+    let asset = getReleaseUrl.data.assets.find(obj => {
+        return re.test(obj.name)
+    })
+
+    const path =  await toolCache.downloadTool(asset.url,
+        undefined,
+        `token ${ghToken}`,
+        {
+            accept: 'application/octet-stream'
+        }
+    );
+
+    core.debug(`Patcher version '${patcherVersion}' has been downloaded at ${path}`);
+    return path
+}
+
+function validateCommand(command) {
+    switch (command) {
+        case "":
+        case "update":
+            return updateCommand;
+        case "report":
+            return reportCommand;
+        default:
+            core.setFailed("Unsupported command - only 'update' and 'report' are supported.");
+            return;
+    }
+}
+
+function updateArgs(updateStrategy, dependency, workingDir) {
+    let args = ["update", noColorFlag, nonInteractiveFlag, skipContainerFlag];
+    if (updateStrategy !== "") {
+        args = args.concat(`--update-strategy=${updateStrategy}`)
+    }
+
+    if (dependency !== "") {
+        args = args.concat(`--target=${dependency}`)
+    }
+
+    return args.concat([workingDir]);
+}
+
+function execOpts(token) {
```

Comment:
Perhaps this function should be called something like `getPatcherEnvVars` and it should just return: 

```
{
            "GITHUB_OAUTH_TOKEN": token,
            "PATCHER_TELEMETRY_ID": telemetryId,
            "HOME": "."
        }
```

then, when you use it it would look like:

```
await exec.getExecOutput(binaryPath,
            [command, nonInteractiveFlag, patcherWorkingDir],
            {env: getPatcherEnvVars(token)})
```

Reply:
Yeah, that's neat. Fixed!